ui/apps/dashboard/src/components/Onboarding/useOnboardingTracking.ts

```diff
@@ -47,5 +47,60 @@ export function useOnboardingTracking() {
     });
   };
 
-  return { trackOnboardingStepCompleted, trackWidgetDismissed, trackOnboardingOpened };
+  const trackCreateAppAction = (type: string) => {
+    trackEvent({
+      name: 'onboarding/create-app.action',
+      data: {
+        type: type,
+      },
+      user: trackingUser,
+      v: '2024-11-04.1',
+    });
+  };
+
```

Comment:
The only question I have is whether we should change this to be more generic like:
```js
{
  name: "onboarding/step.completed",
  data: {
    step: "create-app",
    order: 0, // 1 or zero indexed step #
    isFinalStep: false, // boolean so if we change order or add future steps we can know the user completed the final step
    metadata: {
      hostingProvider: 'xyz'
    }
  },
  user: trackingUser,
}
```

From an analytics perspective, it may be easier over the long haul to have a single event name with many parameters so it can evolve over time.

Are there other ideas or thoughts on the multi-events? I'm open minded, just thinking about it now.

Reply:
🤔 We are tracking 2 different things - steps being completed and clicks. Sometimes clicks lead to steps being completed, sometimes they don't. I'd keep them as two separated events, meaning that we have two event names (onboarding/step.completed and onboarding/click). But it's not a bad idea to keep 1) their data structure more similar, 2) make all the click events the same. Let me make some tweaks to this!